nautobot/ipam/models.py

```diff
@@ -1097,19 +1097,21 @@ def clean(self):
         if self._namespace is None:
             raise ValidationError({"parent": "Either a parent or a namespace must be provided."})
 
-        closest_parent = self._get_closest_parent()
-        # Validate `parent` can be used as the parent for this ipaddress
-        if self.parent:
-            if self.parent != closest_parent:
-                raise ValidationError(
-                    {
-                        "parent": (
-                            f"{self.parent} cannot be assigned as the parent of {self}. "
-                            f" In namespace {self._namespace}, the expected parent would be {closest_parent}."
-                        )
-                    }
-                )
-            self.parent = closest_parent
+        # Host and maxlength are required to get the closest_parent
+        if self.host and self.mask_length:
```

Comment:
Careful - both `host` and `mask_length` can be zero (i.e. falsey), can't they? `0.0.0.0/0`.